src/viam/sdk/tests/test_robot.cpp

```diff
@@ -155,14 +177,20 @@ BOOST_AUTO_TEST_CASE(test_get_status) {
 }
 
 BOOST_AUTO_TEST_CASE(test_get_frame_system_config) {
-    robot_client_to_mocks_pipeline(
-        [](std::shared_ptr<RobotClient> client, MockRobotService& service) -> void {
-            auto mock_fs_config = mock_config_response();
-            auto fs_config = client->get_frame_system_config();
-
-            BOOST_TEST(vec_to_string_util(mock_fs_config) == vec_to_string_util(fs_config),
-                       boost::test_tools::per_element());
-        });
+    robot_client_to_mocks_pipeline([](std::shared_ptr<RobotClient> client,
+                                      MockRobotService& service) -> void {
+        auto mock_fs_config = mock_config_response();
+        auto fs_config = client->get_frame_system_config();
+
+        std::vector<viam::robot::v1::FrameSystemConfig> fs_config_proto;
+        for (const frameSystemConfig& f : fs_config) {
+            fs_config_proto.push_back(f.to_proto());
+        }
+
+        BOOST_TEST(
+            proto_vec_to_string_util(mock_fs_config) == proto_vec_to_string_util(fs_config_proto),
+            boost::test_tools::per_element());
+    });
```

Comment:
So there's a kind of interesting change here (and in the other tests in this file). Previously we were just testing that, e.g., `get_frame_system_config()` worked properly. But now we're also testing the proto conversion! This is definitely good, but it seems a bit half-complete in that, while we're testing the `to_proto` method, we don't seem to be testing the `of_proto` method (at least not directly).

What I'd suggest is a separation of testing concerns here. As per an above comment, we can update the return type of the `mock_{type}_response()` methods so that no conversion is necessary here in `test_robot`. Then, we can add tests for the `{to|from}_proto` methods to the `test_types.cpp` file.

Reply:
Update: I just saw your comment in the description 
> A lot of the testing was done using protos. I was getting errors when trying to compare custom-defined types so continued using protos to compare using the vec_to_string() method. This, arguably, also tests out the to_proto() methods. Should we keep as is or should we change things up? Thought there should be more conversation around this before moving ahead.

I believe my comment speaks to your question @purplenicole730 (let me know if it's not a comprehensive answer!), but just to add an extra note: if there are errors when comparing our types directly then definitely we should investigate and figure out why before merging this.